File: WW1 code/Java Classes/EditBiography.java
import java.sql.SQLException;

public class EditBiography {
    private String surname;
    private String forname;
    private String regiment;
    private int Service_Number;
    private String Biography_attachment;

    public EditBiography() throws SQLException {
    }


    public String getSurname(){
        return surname;
    }
    public void setSurname(String surname){
        this.surname =surname;
    }

    public String getForename(){
        return forname;
    }
    public void setForname(String forname){
        this.forname =forname;
    }

    public String getRegiment(){
        return regiment;
    }
    public void setRegiment(String regiment){
        this.regiment =regiment;
    }

    public int getService_Number() {
        return Service_Number;
    }

    public void setService_Number(int service_Number) {
        Service_Number = service_Number;
    }

    public String getBiography_attachment() {
        return Biography_attachment;
    }

    public void setBiography_attachment(String biography_attachment) {
        Biography_attachment = biography_attachment;
    }

    //public void HTML(Http)
    //de donde saca el soldier service number?
    //int id = Integer.parseInt(request.getParameter("id"));
    EditBiography editBiography = new EditBiography();

    public EditBiography getEditBiography(){
        editBiography.setSurname(surname);
        editBiography.setForname(forname);
        editBiography.setRegiment(regiment);
        editBiography.setService_Number(Service_Number);
        editBiography.setBiography_attachment(Biography_attachment);
        return editBiography;
    }



    EditDAO editDAO = new EditDAO();
    boolean success = editDAO.EditBiography(editBiography);


}

File: WW1 code/PHP code testing/EditBiography.php
<?php
class EditBiography {
    private $surname;
    private $forename;
    private $regiment;
    private $service_Number;
    private $biography_attachment;

    public function __construct() {
        // Constructor vacío, sin excepciones SQL en PHP
    }

    // Getters y Setters
    public function getSurname() {
        return $this->surname;
    }
    public function setSurname($surname) {
        $this->surname = $surname;
    }

    public function getForename() {
        return $this->forename;
    }
    public function setForename($forename) {
        $this->forename = $forename;
    }

    public function getRegiment() {
        return $this->regiment;
    }
    public function setRegiment($regiment) {
        $this->regiment = $regiment;
    }

    public function getService_Number() {
        return $this->service_Number;
    }
    public function setService_Number($service_Number) {
        $this->service_Number = $service_Number;
    }

    public function getBiography_attachment() {
        return $this->biography_attachment;
    }
    public function setBiography_attachment($biography_attachment) {
        $this->biography_attachment = $biography_attachment;
    }

    // Debug function to output all properties
    public function debugOutput() {
        return [
            'surname' => $this->surname,
            'forename' => $this->forename,
            'regiment' => $this->regiment,
            'service_Number' => $this->service_Number,
            'biography_attachment' => $this->biography_attachment
        ];
    }

    // Método para guardar el objeto a través del DAO (se asume que existe la clase EditDAO)
    public function save() {
        $editDAO = new EditDAO();
        return $editDAO->editBiography($this);
    }
}
